src/LibraryViewExtension/Handlers/NodeItemDataProvider.cs

```diff
@@ -35,10 +38,50 @@ class LoadedTypeData
     class NodeItemDataProvider : ResourceProviderBase
     {
         private NodeSearchModel model;
+        private IEventController controller;
+        private Timer throttle;
+        private long throttlingTime; //in millseconds
+        private List<string> items = new List<string>();
 
-        public NodeItemDataProvider(NodeSearchModel model) : base(false)
+        /// <summary>
+        /// Creates a NodeItemDataProvider
+        /// </summary>
+        /// <param name="model">The NodeSearcModel from where nodes data is fetched</param>
+        /// <param name="controller">Event controller for notification</param>
+        /// <param name="throttlingTime">Throttling time in milliseconds</param>
+        public NodeItemDataProvider(NodeSearchModel model, IEventController controller = null, long throttlingTime = 200) : base(false)
         {
             this.model = model;
+            if (model == null) throw new ArgumentNullException("model");
+
+            if(controller != null)
+            {
+                model.EntryAdded += OnLibraryDataUpdated;
+                model.EntryRemoved += OnLibraryDataUpdated;
+                model.EntryUpdated += OnLibraryDataUpdated;
+
+                this.controller = controller;
+                this.throttlingTime = throttlingTime;
+                throttle = new Timer(RaiseLibraryDataUpdated, controller, Timeout.Infinite, Timeout.Infinite); //disabled at begining
+            }
+        }
+
+        private void OnLibraryDataUpdated(NodeSearchElement obj)
+        {
+            items.Add(FullyQualifiedName(obj));
+            //Raise event only after due milliseconds.
+            throttle.Change(throttlingTime, 0); //enabled now
+        }
+
+        private void RaiseLibraryDataUpdated(object state)
+        {
+            if(controller != null)
+            {
+                var text = string.Join(", ", items);
+                controller.RaiseEvent("libraryDataUpdated", text);
+                //reset items
+                items.Clear();
```

Comment:
There is a chance of risk condition for items. See if at the same time when the timer callback is invoked, new entries are added/removed/updated.